app/src/main/java/mozilla/lockbox/adapter/SectionedAdapter.kt

```diff
@@ -0,0 +1,115 @@
+/*
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+// Large portions of this class are borrowed from:
+// https://gist.github.com/gabrielemariotti/4c189fb1124df4556058
```

Comment:
@linuxwolf do you have any thoughts here?

Reply:
I see someone asked the author about licensing once already...

Would you all mind trying to reach out via another GitHub comment and perhaps email (if available/findable)? It definitely gets a bit sticky here but the "spirit" of the sharing of the code is definitely for others to re-use it.

Reply:
The lack of a license is concerning; we might need to include someone more knowledgeable on licensing.

Generally, though, code (heavily or lightly) inspired by another codebase can be relicensed if the original source license is compatible (for MPL, that's just about everything).  Since the license is not stated in the gist or its included files, was there more information from sources that led to this gist (and most importantly, are they stable enough to use as a reference)?

Reply:
>Since the license is not stated in the gist or its included files, was there more information from sources that led to this gist

I investigated a bit more myself and the author has not explicitly explained the license situation. I've added this to our legal review checklist: #161